### PR TITLE
Remove Gulp, switch to TypeScript, Unit Tests.

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,20 +40,36 @@
...
-    "jshint": "jshint index.js --reporter=jslint",
-    "test": "npm -s run-script jshint && mocha -t 50000",
-    "gulp": "gulp"
+    "dn.clean": "dotnet clean openapi-diff/OpenApiDiff.sln",
+    "dn.restore": "dotnet restore openapi-diff/OpenApiDiff.sln",
+    "dn.build": "npm run dn.clean && npm run dn.restore && dotnet build -c release openapi-diff/OpenApiDiff.sln /nologo /clp:NoSummary",
+    "dn.publish": "npm run dn.build && dotnet publish -c release openapi-diff/src/core/OpenApiDiff/OpenApiDiff.csproj",
+    "dn.test": "npm run dn.build && dotnet test openapi-diff/src/modeler/AutoRest.Swagger.Tests/AutoRest.Swagger.Tests.csproj",
+    "tsc": "tsc",
+    "ts.test": "tsc && jest",
```
Comment (vladbarosan) [inline package.json]: these are a little confusing, not sure which one i need to run to validate a change :)

Comment (sergey-shandar) [reply package.json]: Always run `test` to test everything. Nothing is changed. :-)

File: src/lib/util/utils.ts [hunk 2]
```diff
@@ -447,17 +463,28 @@ exports.getJsonContentType = function getJsonContentType(consumesOrProduces) {
...
+  }
+}
+
 /**
  * Determines whether the given model is a pure (free-form) object candidate (i.e. equivalent of the C# Object type).
  * @param {object} model - The model to be verified
  * @returns {boolean} result - true if model is a pure object; false otherwise.
  */
-exports.isPureObject = function isPureObject(model) {
+export function isPureObject(model: Model) {
```
Comment (veronicagg) [inline src/lib/util/utils.ts]: Not for this PR, but given the code that's been copied from oav, we should extract out and share, seems like this tool needs a bunch of restructuring.

Comment (sergey-shandar) [reply src/lib/util/utils.ts]: I'm considering to create a package for OpenAPI 2.0 types and parsing.

**Pull request merged**